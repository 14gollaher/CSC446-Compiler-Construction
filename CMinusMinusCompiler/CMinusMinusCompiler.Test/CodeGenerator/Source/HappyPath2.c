int calculateSum(int x, int y)
{
	int a, b;
	a = x;
	b = y;
	return a + b;
}

int main()
{
	int x, y;
	x = -1;
	y = 5;
	x = calculateSum(x, y);
	cout << "The sum is: " << x;
	return -1;
}

