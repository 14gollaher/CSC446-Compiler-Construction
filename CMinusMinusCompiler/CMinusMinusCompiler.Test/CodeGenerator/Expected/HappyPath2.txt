.MODEL SMALL    
.STACK 100H     
                
.DATA           
__S0 DB "The sum is: ", "$" 
                
.CODE           
INCLUDE io.asm  
_calculateSum   PROC
                PUSH BP
                MOV BP, SP
                SUB SP, 6
                
                MOV AX, [BP+4]
                MOV [BP-2], AX
                
                MOV AX, [BP+6]
                MOV [BP-4], AX
                
                MOV AX, [BP-2]
                MOV BX, [BP-4]
                ADD AX, BX
                MOV [BP-6], AX
                
                MOV AX, [BP-6]
                MOV AX, AX
                
                ADD SP, 6
                POP BP
                RET 4
_calculateSum   ENDP
                
                
_main           PROC
                PUSH BP
                MOV BP, SP
                SUB SP, 10
                
                MOV AX, 1
                NEG AX
                MOV [BP-6], AX
                
                MOV AX, [BP-6]
                MOV [BP-2], AX
                
                MOV AX, 5
                MOV [BP-8], AX
                
                MOV AX, [BP-8]
                MOV [BP-4], AX
                
                PUSH [BP-4]
                
                PUSH [BP-2]
                
                CALL _calculateSum
                
                MOV AX, AX
                MOV [BP-2], AX
                
                MOV DX, OFFSET __S0
                CALL writestr
                
                MOV AX, [BP-2]
                CALL writeint
                
                MOV AX, 1
                NEG AX
                MOV [BP-10], AX
                
                MOV AX, [BP-10]
                MOV AX, AX
                
                ADD SP, 10
                POP BP
                RET 0
_main           ENDP
                
                
__STARTPROC     PROC
                MOV AX, @data
                MOV DS, AX
                
                CALL _main
                
                MOV AX, 4C00H
                INT 21H
__STARTPROC     ENDP
END __STARTPROC 
